pkg/container/vector/vector.go

```diff
@@ -21,46 +21,39 @@ import (
 	"unsafe"
 
 	"github.com/matrixorigin/matrixone/pkg/common/bitmap"
+	"github.com/matrixorigin/matrixone/pkg/vectorize/shuffle"
 
 	"github.com/matrixorigin/matrixone/pkg/common/moerr"
 	"github.com/matrixorigin/matrixone/pkg/common/mpool"
 	"github.com/matrixorigin/matrixone/pkg/container/nulls"
 	"github.com/matrixorigin/matrixone/pkg/container/types"
-	"github.com/matrixorigin/matrixone/pkg/vectorize/shuffle"
 )
 
-// XXX Moved vector from types.go to vector.go
-// XXX Deleted vector interface, which was commented out and outdated anyway.
-/* Vector vector
- * origin true:
- * 				count || type || bitmap size || bitmap || vector
- * origin false:
- *  			count || vector
- */
+const (
+	FLAT     = iota // flat vector represent a uncompressed vector
+	CONSTANT        // const vector
+	DIST            // dictionary vector
+)
+
+// Vector represent a column
 type Vector struct {
-	// XXX There was Ref and Link, from the impl, it is totally wrong stuff.
-	// Removed.
-	Typ types.Type
-	Col interface{}  // column data, encoded Data
-	Nsp *nulls.Nulls // nulls list
+	// vector's class
+	class int
+	// type represent the type of column
+	typ types.Type
+	nsp *nulls.Nulls // nulls list
```

Comment:
nulls.Nulls, do we really need another level of abstraction?  can it be just a bitmap?  

now this is a pointer to Nulls, which itself is a pointer to bitmap, do we really need ptr to ptr?